### ISSUE TITLE
Support customization/localization of reports

### ISSUE THREAD

**Issue by mildred** [simonmichael/hledger#1025]
Reports (such as balance sheet or income statement) should be able to be customized so they can be localized and translated.

Some simple words like Assets or Liabilities must be customized (in French, it corresponds to *Actif* and *Passif*)

Split from #1006

**Comment by simonmichael** [comment]
Agreed! Thanks for the issue. We need someone to research and/or build a small prototype, eg localising just the title of the balancesheet report. Also, to think about the hledger as a whole and clarify which parts can/should be localised. Could you help ?

**Comment by simonmichael** [comment]
Related: #1219

**Comment by simonmichael** [comment]
Related: #1219

**Comment by simonmichael** [comment]
Perhaps implementing the flexible `balancereport` command would be a good start. Help welcome, including a better name for it.

**Comment by simonmichael** [comment]
Open questions:

- in current compound balance reports, each subsection shows an account type. In the mockups above they are based on arbitrary queries, and account types aren't mentioned. What would account types be used for ?
- the above doesn't really address how to localise headings of the standard builtin reports, unless we allowed custom commands to override builtin commands, and it would still be a bit over-complex and error-prone.
- it would be useful to be able to localise the standard report titles once eg in a user config file, and not have to do it in every journal or on every command line.
- it could be useful to start doing traditional l10n, building in translations of the standard report titles and selecting them based on system locale with no configuration needed. There would need to be a way to override this.

**Comment by alerque** [comment]
Also things like month names should follow the output locale:

```
$ LANG=tr_TR.UTF-8 hledger -f=maliye.ledger bal Daire --transpose --no-total --empty -M
     ||  Daire 1   Daire 2   Daire 3   Daire 4   Daire 5
=====++==================================================
 Jan || ₺-250,00         0         0  ₺-250,00         0
 Feb || ₺-250,00         0         0  ₺-250,00         0
 Mar || ₺-250,00         0         0  ₺-250,00         0
 Apr || ₺-250,00  ₺-250,00  ₺-250,00  ₺-250,00  ₺-250,00
```

Should be:

```
     ||  Daire 1   Daire 2   Daire 3   Daire 4   Daire 5
=====++==================================================
 Oca || ₺-250,00         0         0  ₺-250,00         0
 Şub || ₺-250,00         0         0  ₺-250,00         0
 Mar || ₺-250,00         0         0  ₺-250,00         0
 Nis || ₺-250,00  ₺-250,00  ₺-250,00  ₺-250,00  ₺-250,00
 ```

Also a CLI flag to turn off the month name substitutions and include dates (as seen in `-O csv` for the same query) might be in order no matter the state of languages.

 So is there a Project Fluent library for Haskell?
